Ruby/Ruby.sublime-syntax

```diff
@@ -330,22 +329,77 @@ contexts:
-      scope: keyword.control.ruby
-    - match: '\b(class|module)\b(?![?!])'
-      scope: keyword.control.ruby
-    - match: '\b(begin|end|ensure|rescue)\b(?![?!])'
-      scope: keyword.control.ruby
-    - match: '\b(case|else|ensure|for|in|then)\b(?![?!])'
-      scope: keyword.control.ruby
-    - match: '\b(elsif|if|unless|when|while|until)\b(?![?!])'
-      scope: keyword.control.ruby
+    # blocks
+    - match: \bbegin\b(?![?!])
+      scope: keyword.control.block.begin.ruby
+      push: after-keyword
+    - match: \bend\b(?![?!])
+      scope: keyword.control.block.end.ruby
+    # conditional
+    - match: \bcase\b(?![?!])
+      scope: keyword.control.conditional.case.ruby
+      push: after-keyword
+    - match: \belse\b(?![?!])
+      scope: keyword.control.conditional.if.ruby
+      push: after-keyword
+    - match: \belsif\b(?![?!])
+      scope: keyword.control.conditional.elseif.ruby
+      push: after-keyword
+    - match: \bif\b(?![?!])
+      scope: keyword.control.conditional.if.ruby
+      push: after-keyword
+    - match: \bthen\b(?![?!])
+      scope: keyword.control.conditional.then.ruby
+    - match: \bunless\b(?![?!])
+      scope: keyword.control.conditional.unless.ruby
+      push: after-keyword
+    - match: \bwhen\b(?![?!])
+      scope: keyword.control.conditional.when.ruby
+      push: after-keyword
+    # exception
+    - match: \bensure\b(?![?!])
+      scope: keyword.control.exception.ensure.ruby
+      push: after-keyword
+    - match: \brescue\b(?![?!])
+      scope: keyword.control.exception.rescue.ruby
+      push: after-keyword
+    # loop
+    - match: \bfor\b(?![?!])
+      scope: keyword.control.loop.for.ruby
+      push: after-keyword
+    - match: \buntil\b(?![?!])
+      scope: keyword.control.loop.until.ruby
       push: after-keyword
-    - match: \b(and|not|or)\b
+    - match: \bwhile\b(?![?!])
+      scope: keyword.control.loop.while.ruby
+      push: after-keyword
+    # flow
+    - match: \bbreak\b(?![?!])
+      scope: keyword.control.flow.break.ruby
+    - match: \bnext\b(?![?!])
+      scope: keyword.control.flow.next.ruby
+    - match: \bredo\b(?![?!])
+      scope: keyword.control.flow.redo.ruby
+    - match: \bretry\b(?![?!])
+      scope: keyword.control.flow.retry.ruby
+    - match: \breturn\b(?![?!])
+      scope: keyword.control.flow.return.ruby
+    - match: \bsuper(?![?!])
+      scope: keyword.control.flow.super.ruby
+    - match: \byield\b(?![?!])
+      scope: keyword.control.flow.yield.ruby
+    # operators
+    - match: \b(?:and|not|or)\b
+      scope: keyword.operator.logical.ruby
+      push: after-keyword
+    - match: \bin\b(?![?!])
       scope: keyword.operator.logical.ruby
       push: after-keyword
-    - match: '\b(alias|alias_method|break|next|redo|retry|return|super|undef|yield)\b(?![?!])|\bdefined\?|\bblock_given\?'
-      scope: keyword.control.pseudo-method.ruby
+    # other
+    - match: \b(?:BEGIN|END)\b(?![?!])
+      scope: entity.name.function.prepocessor.ruby
```

Comment:
What's this?

Reply:
This is a feature borrowed from Perl, I think. It is a feature to hack the preprocessor/import of an module. In perl it can be used to manipulate the lookup path of imports or the whole environment. 

The scope is borrowed from:

https://github.com/sublimehq/Packages/blob/f60f80afdb66e317c3f2a0852861171ab677621f/Perl/Perl.sublime-syntax#L461-L465

Reply:
https://ruby-doc.org/docs/ruby-doc-bundle/Manual/man-1.4/syntax.html#BEGIN_proc

Reply:
... sorry for being too lazy to create a dedicated context for them as in Perl at this point.

Reply:
Thanks for the reference. The scope name is fine, then.

Reply:
Possible due to their special meaning. But basically, they are ordinary functions.